### ISSUE TITLE
BatchNorm: Make running mean/std estimation optional

### ISSUE THREAD

**Issue by berleon** [fchollet/keras#1885]
Introduce a new constructor argument `update_averages`. If it is set, then there are no updates to the `running_mean` and `running_std` members.

If this change sound good to you, I would get a pull request ready.

**Comment by philipperemy** [comment]
Can you provide a use case of when it could be useful to add another constructor to BatchNormalization?

That will help.

Thanks.

**Comment by berleon** [comment]
When you don't have a test phase, e.g. you only evaluate your model at the end of a long training phase. The running_mean/std can then be estimated during the last epoch. My special use-case are GANs, they don't have a test time.

**Comment by fchollet** [comment]
I don't think that would be widely useful, at least not enough justify inclusion in the codebase. But if it is very easy to add to your own fork of Keras, for your own needs.
